src/Thread/BoundedBlockingBuffer.java

```diff
@@ -0,0 +1,44 @@
+package Thread;
+
+public class BoundedBlockingBuffer<T> {
+    T data;
+    boolean isEmpty = true;
+
+    public synchronized T take() {
+        while (isEmpty) {
```

Comment:
Take method, you can write like this:

 while (!isEmpty) {
            try {
                wait();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
        isEmpty = false;
        System.out.println("Take " +this.data);
        notify();
        return this.data;
    }